### PR TITLE
12763 Split PublicKeyStorageProvider

### LINKED ISSUE #12763: Split `PublicKeyStorageProvider`
## Description

`PublicKeyStorageProvider` contains method `clearCache` which is tied to legacy store. This needs to be extracted from that interface and moved to legacy store together with the respective endpoint (`RealmAdminResource.clearKeysCache()`).

`InfinispanPublicKeyStorageProviderFactory` should be made `EnvironmentDependentProviderFactory` conditioned on `MAP_STORAGE` disabled.

Implement a simplistic map-storage implementation of `PublicKeyStorageProvider` which will always load key from the respective `PublicKeyLoader`.

### PR DESCRIPTION
**Pull request by martin-kanis**
Closes #12763

### CODE REVIEW

File: testsuite/integration-arquillian/tests/base/src/test/resources/META-INF/keycloak-server.json [hunk 1]
```diff
@@ -209,6 +209,12 @@
         }
     },
```
Comment (sguilhen) [inline testsuite/integration-arquillian/tests/base/src/test/resources/META-INF/keycloak-server.json:211]: Shouldn't the `publicKeyStorage` config be present here too?

Comment (martin-kanis) [reply testsuite/integration-arquillian/tests/base/src/test/resources/META-INF/keycloak-server.json:220]: You are right, added.

**Review by sguilhen — APPROVED** [review]

**Review by pedroigor — APPROVED** [review]

**Comment by pedroigor** [comment]
@martin-kanis LGTM. Could you rebase, please?

**Comment by ahus1** [comment]
As this PR gathered conflicts and it should be merged soon, there is now a rebased PR here: #12897

**Pull request closed without merging**